src/vs/workbench/services/progress/browser/progressIndicator.ts

```diff
@@ -194,17 +198,23 @@ export class CompositeProgressIndicator extends CompositeScope implements IProgr
 		progressbar: ProgressBar,
 		scopeId: string,
 		isActive: boolean,
+		private readonly options: { hideWhenInactive?: boolean } | undefined,
```

Comment:
@eamodio everything is fine for me, but this option is impossible to understand, because it is in fact not just used to hide the progress when the view becomes deactivated but also seems to be used in other places. since I do not understand it and do not really like it, I am against this change because it makes this component harder to use. please find an alternative maybe by aligning all progress indicators to behave the same.

Reply:
The (poorly named) option, is basically a flag to signal that the progress bar for the indicator is shared or not. If it is shared (e.g. viewlet), then we only want to hide the progress if it is active. If it is not shared (e.g. view), then we want to stop it in any of those cases.

Reply:
Maybe a better name is `hideEvenWhenInactive` or `exclusiveProgressBar`

Reply:
@eamodio how is the progress indicator for a view different for a viewlet or panel? I do not understand it nor the concept of a shared indicator.

My understanding is that a progress indicator has a scope (view id, viewlet id or panel id) and only reacts on events matching this ID. So if a viewlet changes, we update the progress bar in the same way I would think we should do it for a view. How is a view different?

If at all we do figure out that we need to have different behaviour, maybe we could have a base class for the indicator and then 2 specific cases: shared and non-shared indicator. But again, I see no reason yet to go there.

Reply:
https://github.com/microsoft/vscode/pull/92136#issuecomment-595799821 seems relevant, and we need to understand it better I guess.

Reply:
The root difference is that all viewlets share a single progress bar, since its always in the same location at the top of the sidebar and there can only be one progress showing at a time. Where as the view each have their own progress bar which call have multiple showing at any time.

Reply:
@eamodio I get that and still: when the viewlet is inactive, I want the progress to stop. same as I would expect that for a view that is hidden or collapsed.

Reply:
Not really, because the viewlet that became active could have running progress which will not use that same progress bar, so if we stopped it, it could interfere with it. But again, I didn't create, nor change that behavior -- that was the existing behavior. I just needed to adapt it because that doesn't work for the view case where there are independent progress bars.